cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/co/cask/cdap/etl/common/PipelineRegisterer.java

```diff
@@ -156,7 +164,107 @@ public Pipeline registerPlugins(ETLConfig config, Class errorDatasetType, Datase
       throw new RuntimeException(e);
     }
 
-    return new Pipeline(sourcePluginId, sinksInfo, transformInfos);
+    Map<String, List<String>> connectionsMapString =
```

Comment:
isn't this map eventually serialized?  Maps.transformEntries applies the transform lazily so isn't serializable.